### PR TITLE
SINGA -475 add Sub operator implementation to singa

### PR DESCRIPTION
**Pull request by pinpom**

### CODE REVIEW

File: test/python/test_operation.py [hunk 1]
```diff
@@ -322,6 +322,21 @@ def test_LeakyRelu(self):
...
+        x = tensor.from_numpy(X)
+        y = tensor.from_numpy(Y)
+        x.to_device(gpu_dev) 
+        y.to_device(gpu_dev) 
+
+        result=autograd.sub(x,y)
+        dx=result.creator.backward(x.data)[0]  
+
+        result_np=np.subtract(X,Y)      
+        np.testing.assert_array_almost_equal(tensor.to_numpy(result), result_np)
```
Comment (joddiy) [inline test/python/test_operation.py]: please add the test case of backward propagation.

Comment (pinpom) [reply test/python/test_operation.py]: added backward test.

File: python/singa/autograd.py [hunk 2]
```diff
@@ -1828,3 +1828,21 @@ def backward(self, dy):
 
 def leakyrelu(x, a=0.01):
     return LeakyRelu(a)(x)[0]
+
+
+class Sub(Operation):
+    def __init__(self):
+        super(Sub, self)._init_()
```
Comment (chrishkchris) [inline python/singa/autograd.py]: line1835:      super(Sub, self)._init_()  
should be
 super(Sub, self).__init__()

Comment (pinpom) [reply python/singa/autograd.py]: fix the code

**Comment by chrishkchris** [comment]
Tested on cloud K80 with the code. Ready for merge

**Pull request merged**